patient_data_source.py

```diff
@@ -0,0 +1,214 @@
+from abc import ABC, abstractmethod
+import names
+import warnings
+from datetime import datetime
+from enum import Enum
+from collections.abc import Iterable
+from fhirclient.models.patient import Patient as FHIR_Patient
+from fhirclient.models.observation import Observation as FHIR_Observation
+from fhirclient.models.fhirdate import FHIRDate
+
+class Patient(ABC):
+	class Gender(Enum):
+		male = 0
+		female = 1
+		other = 2
+		unknown = 3
+
+	def get_gender(self) -> Gender:
+		"""Returns the patient's gender. Default implementation return Gender.unkown"""		
+		return Gender.unknown
+
+	def get_identifier_value(self) -> str:
+		"""Returns a id value for external use."""
+		return None
+
+	@abstractmethod
```

Comment:
this is not abstractmethod anymore now that it has a default implementation, right?